--- a/src/helpers/Format.cpp
+++ b/src/helpers/Format.cpp
@@ -314,13 +314,15 @@

 std::string NFormatUtils::drmFormatName(DRMFormat drm) {
     auto        n    = drmGetFormatName(drm);
-    std::string name = n;
-    free(n); // NOLINT(cppcoreguidelines-no-malloc,-warnings-as-errors)
+    std::string name = n ? n : "UNKNOWN";
+    if (n)
+        free(n); // NOLINT(cppcoreguidelines-no-malloc,-warnings-as-errors)
     return name;
 }

 std::string NFormatUtils::drmModifierName(uint64_t mod) {
     auto        n    = drmGetFormatModifierName(mod);
-    std::string name = n;
-    free(n); // NOLINT(cppcoreguidelines-no-malloc,-warnings-as-errors)
+    std::string name = n ? n : "UNKNOWN";
+    if (n)
+        free(n); // NOLINT(cppcoreguidelines-no-malloc,-warnings-as-errors)
     return name;
 }
