--- a/helpers.py
+++ b/helpers.py
@@ -7,6 +7,7 @@ from types import MappingProxyType
 import operator
 import json
 import subprocess
+import shutil
 import logging

 Prev = TypeVar("Prev")
@@ -63,7 +64,7 @@ def read_system_theme():
     try:
         output = subprocess.check_output(
             [
-                "/usr/bin/busctl",
+                shutil.which("busctl"),
                 "--user",
                 "--json=short",
                 "call",
--- a/main.py
+++ b/main.py
@@ -121,7 +121,7 @@ def monitor():
     current_mode = read_system_theme()
     with subprocess.Popen(
         [
-            "/usr/bin/busctl",
+            shutil.which("busctl"),
             "--user",
             "--json=short",
             "--match",
