diff --git a/helpers.py b/helpers.py
index 1234567..abcdefg 100644
--- a/helpers.py
+++ b/helpers.py
@@ -6,6 +6,7 @@ from types import MappingProxyType
 import operator
 import json
 import subprocess
+import shutil
 import logging

 Prev = TypeVar("Prev")
@@ -63,7 +64,7 @@ def read_system_theme():
     try:
         output = subprocess.check_output(
             [
-                "/usr/bin/busctl",
+                shutil.which("busctl"),
                 "--user",
                 "--json=short",
                 "call",
diff --git a/main.py b/main.py
index 1234567..abcdefg 100644
--- a/main.py
+++ b/main.py
@@ -38,16 +38,8 @@ class AutoDarkLinuxInputHandler(sublime_plugin.ListInputHandler):
     def placeholder(self):
-        return "Choose mode"
+        return "Color scheme mode"

-    def description(self, _: str, text: str):
-        # TODO: Test this
-        return f"{text} Mode"
-
-    def list_items(self, options=[("Light", "light"), ("Dark", "dark"), ("System", "system")]):
-        # TODO: Test this
-        mode = plugin_settings.get("auto_dark_mode", "system")
-        index = next((i for i, v in enumerate(options) if v[1] == mode.lower()))
-        return (options, index)
+    def list_items(self):
+        return [("Dark", "dark"), ("Light", "light"), ("System", "system")]


 class AutoDarkLinuxCommand(sublime_plugin.ApplicationCommand):
@@ -90,7 +82,6 @@ def plugin_loaded():
         return sublime.message_dialog(
             "AutoDarkLinux plugin requires the 'busctl' command from systemd"
         )
-    sublime.save_settings(plugin_settings_file)
     global plugin_settings
     plugin_settings = sublime.load_settings(plugin_settings_file)

@@ -121,7 +112,7 @@ def monitor():
     current_mode = read_system_theme()
     with subprocess.Popen(
         [
-            "/usr/bin/busctl",
+            shutil.which("busctl"),
             "--user",
             "--json=short",
             "--match",
